File: ReservationSystem-Server/ReservationSystem-Server/Data/Restaurant.cs
namespace ReservationSystem_Server.Data;

public class Restaurant
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Address { get; set; } = null!;
    public string PhoneNumber { get; set; } = null!;
    public string Email { get; set; } = null!;

    public List<RestaurantArea> Areas { get; set; } = new();
    public List<Sitting> Sittings { get; set; } = new();
}

File: reservation-system-app/services/api/restaurant.js
import common from "./common";
import {createContext} from "react";

/**
 * @typedef {Object} Restaurant
 * @property {string} name
 * @property {string} address
 * @property {string} phoneNumber
 * @property {string} email
 *
 * @type {React.Context<Restaurant>}
 */
const RestaurantContext = createContext(null);

/**
 * @returns {Promise<* | ErrorDesc>}
 */
async function getRestaurant() {
    const response = await common.fetch(`restaurant`, "GET");

    if (response.ok) {
        return await response.json();
    }

    return await common.processError(response);
}

export default {
    getRestaurant, RestaurantContext
};
